### ISSUE TITLE
[Format] Why are there so many seemingly pointless timelines?

### ISSUE THREAD

**Issue by iboB** [EsotericSoftware/spine-runtimes#542]
When exporting a json file there are many timelines that seem to do nothing (but just take cpu). There are many like this in spineboy.json

				"scale": [
					{ "time": 0, "x": 1, "y": 1, "curve": "stepped" },
					{ "time": 0.8, "x": 1, "y": 1 }
				]

Am I missing something? This seems to set the scale to 1 at time 0 only to set it to... 1 at time 0.8. What is the point of this?

**Comment by pharan** [comment]
Each Spine Animation is a list of changes. And Spine's (current) default behavior is to not do anything unless you tell it to. This means it won't change anything unless it's in that list of changes. This list is reflected in the Spine dopesheet.

So there is no point to it, unless that value is something you want to lock-in or override from previously applied animations. This includes situations where two animations are playing at the same time.

In all other cases, this stems from a dual problem that used to exist. that (1) the default Spine behavior above caused weird poses because animations aren't guaranteed to have equivalent sets of changes, so some poses persist from previously played animations after you play subsequent ones. Seeing this, you had to (animators assume) key every item even if they do nothing.

But the truth is that the programmer just needs to tell the runtime to self-reset before playing a new animation, or whenever they want resets to happen. A more convenient way to do this is coming in an AnimationState class update soon.

...and (2) it was a bit hard to copy poses from one time and one animation to another, so animators also assumed (rightly) that it was easier to just key every transform value and copy that from point to point. Thankfully, this is no longer the case. The recommended option is to use the Key Dopesheet function `CTRL`+`SHIFT`+`L`. This adds keys only for keyed items in the dopesheet and no more.

So, in short, the point is that it's for overriding. But the rest of the time, it's the animator/programmer's mistake, and those values are there because they were keyed in the editor.

I'm in the middle of writing some documentation on this. It should hopefully be done next week.
